configuration/file_test.go

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package configuration
+
+import (
+	"github.com/01org/ciao/payloads"
+	"github.com/google/gofuzz"
+	"io/ioutil"
+	"syscall"
+	"testing"
+)
+
+// missing the 'configure' identation
+const malformedConf = `---
+scheduler:
+    storage_uri: /etc/ciao/configuration.yaml
+controller:
+    compute_ca: /etc/pki/ciao/compute_ca.pem
+    compute_cert: /etc/pki/ciao/compute_key.pem
+    identity_user: controller
+    identity_password: ciao
+launcher:
+    compute_net: 192.168.1.110
+    mgmt_net: 192.168.1.111
+image_service:
+    url: http://glance.example.com
+identity_service:
+    url: http://keystone.example.com
+`
+
+func TestLoadFile(t *testing.T) {
+	var fuzzyStr string
+	f := fuzz.New()
+	f.Fuzz(&fuzzyStr)
+
+	// create temp file where we can read the conf
+	tmpf, err := ioutil.TempFile("", "configuration.yaml")
+	if err != nil {
+		panic(err)
+	}
+	defer syscall.Unlink(tmpf.Name())
+	ioutil.WriteFile(tmpf.Name(), []byte(minValidConf), 0644)
+
+	_, err = loadFile(tmpf.Name())
+	if err != nil {
+		t.Fatalf("expected nil, got %v", err)
+	}
+
+	yamlConf, _ := loadFile(fuzzyStr)
+	if yamlConf != nil {
+		t.Fatalf("expected nil, got %v", yamlConf)
+	}
+
+}
+
+func TestFileFetchConfiguration(t *testing.T) {
+	var fuzzyStr string
+	var d driver
+
+	f := fuzz.New()
+	f.Fuzz(&fuzzyStr)
+
+	d = &file{}
+
+	// uri parse error due to  "%z " on string
+	_, err := d.fetchConfiguration("%z" + fuzzyStr)
+	if err == nil {
+		t.Fatalf("expected error, got %v", err)
+	}
+
+	// file doesn't exists
+	_, err = d.fetchConfiguration(emptyPathURI + "/" + fuzzyStr)
+	if err == nil {
+		t.Fatalf("expected error, got %v", err)
+	}
```

Comment:
We should do positive testing as well, i.e. veryfy that a valid URI gives you the expected answer.